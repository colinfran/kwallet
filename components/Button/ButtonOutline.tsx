import React, { useContext } from "react"
import { DataContext } from "../../providers/DataProvider"
import { Button } from "native-base"

type ButtonProps = {
  text: string
  route?: string
  isDisabled?: boolean
  leftIcon?: JSX.Element[] | JSX.Element
  onPress?: () => void
  textColor?: string
}

const ButtonOutline = ({
  text,
  isDisabled = false,
  leftIcon,
  onPress,
  textColor,
}: ButtonProps): JSX.Element => {
  const { appColor, textColor: fontColor } = useContext(DataContext)
  return (
    <Button
      _text={{
        style: {
          color: textColor || fontColor,
        },
      }}
      borderColor={appColor}
      borderRadius={15}
      isDisabled={isDisabled}
      leftIcon={leftIcon}
      size="lg"
      testID="test-button"
      variant="outline"
      onPress={onPress}
    >
      {text}
    </Button>
  )
}

export default ButtonOutline
