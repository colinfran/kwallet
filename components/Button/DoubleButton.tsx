import React, { useContext } from "react"
import { StyleSheet, View } from "react-native"
import { DataContext } from "../../providers/DataProvider"
import { Button } from "native-base"

const DoubleButton = ({
  left,
  right,
  style = {},
  pointerEvents = "auto",
  buttonGroupStyle,
  buttonProps,
}): JSX.Element => {
  const { appColor } = useContext(DataContext)

  return (
    <View style={[styles.container, style]}>
      <View>
        <View style={styles.row}>
          <View style={[styles.iconContainer]}>
            <Button.Group
              borderColor="red"
              borderRadius={15}
              // borderWidth={2}
              colorScheme="blue"
              pointerEvents={pointerEvents as any}
              style={buttonGroupStyle}
              variant="outline"
              isAttached
            >
              <Button
                {...buttonProps}
                pointerEvents={pointerEvents as any}
                style={{ borderColor: appColor, borderWidth: 2 }}
                testID="test-left"
                w="50%"
                onPress={left.onPress}
              >
                {left.text}
              </Button>
              <Button
                {...buttonProps}
                pointerEvents={pointerEvents as any}
                style={{ borderColor: appColor, borderWidth: 2 }}
                testID="test-right"
                w="50%"
                onPress={right.onPress}
              >
                {right.text}
              </Button>
            </Button.Group>
          </View>
        </View>
      </View>
    </View>
  )
}

export default DoubleButton

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    borderRadius: 15,
  },
  row: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    gap: 40,
  },
  iconContainer: {
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
  },
})
