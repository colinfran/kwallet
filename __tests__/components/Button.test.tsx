import React from "react"
import DoubleButton from "../../components/Button/DoubleButton"
import { Providers } from "../helper"
import { render, screen, fireEvent } from "@testing-library/react-native"
import "@testing-library/jest-native/extend-expect"
import ButtonOutline from "../../components/Button/ButtonOutline"

const onPressEventFunc1 = jest.fn()
const onPressEventFunc2 = jest.fn()
const onPressEventFunc3 = jest.fn()

describe("Button", () => {
  describe("<DoubleButton />", () => {
    it("renders correctly", () => {
      const tree = render(
        <DoubleButton
          buttonGroupStyle={undefined}
          buttonProps={undefined}
          left={{
            text: "left",
            onPress: onPressEventFunc1,
          }}
          right={{
            text: "right",
            onPress: onPressEventFunc2,
          }}
        />,
        { wrapper: Providers }
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("text", () => {
      render(
        <DoubleButton
          buttonGroupStyle={undefined}
          buttonProps={undefined}
          left={{
            text: "left",
            onPress: onPressEventFunc1,
          }}
          right={{
            text: "right",
            onPress: onPressEventFunc2,
          }}
        />,
        { wrapper: Providers }
      )
      const leftBtn = screen.getByTestId("test-left")
      const rightBtn = screen.getByTestId("test-right")
      expect(leftBtn).toHaveTextContent("left")
      expect(rightBtn).toHaveTextContent("right")
    })

    it("onClick", () => {
      render(
        <DoubleButton
          buttonGroupStyle={undefined}
          buttonProps={undefined}
          left={{
            text: "left",
            onPress: onPressEventFunc1,
          }}
          right={{
            text: "right",
            onPress: onPressEventFunc2,
          }}
        />,
        { wrapper: Providers }
      )
      const leftBtn = screen.getByTestId("test-left")
      const rightBtn = screen.getByTestId("test-right")
      fireEvent.press(leftBtn)
      fireEvent.press(rightBtn)
      expect(onPressEventFunc1.mock.calls.length).toBe(1)
      expect(onPressEventFunc2.mock.calls.length).toBe(1)
    })
  })
  describe("<ButtonOutline />", () => {
    it("renders correctly", () => {
      const tree = render(
        <ButtonOutline text="outlineButton" onPress={onPressEventFunc3} />,
        { wrapper: Providers }
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it("text", () => {
      render(
        <ButtonOutline text="outlineButton" onPress={onPressEventFunc3} />,
        { wrapper: Providers }
      )
      const btn = screen.getByTestId("test-button")
      expect(btn).toHaveTextContent("outlineButton")
    })

    it("onClick", () => {
      render(
        <ButtonOutline text="outlineButton" onPress={onPressEventFunc3} />,
        { wrapper: Providers }
      )
      const btn = screen.getByTestId("test-button")
      fireEvent.press(btn)
      expect(onPressEventFunc3.mock.calls.length).toBe(1)
    })
  })
})
