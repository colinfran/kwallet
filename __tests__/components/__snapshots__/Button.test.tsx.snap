// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button <ButtonOutline /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    colorScheme="primary"
    dataSet={{}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "#123456",
        "borderRadius": 15,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 12,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 12,
      }
    }
    testID="test-button"
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      test={true}
    >
      <View
        dataSet={{}}
        style={{}}
      >
        <Text
          dataSet={{}}
          style={
            [
              {
                "backgroundColor": undefined,
                "color": "#66B0A3",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 24,
                "textDecorationLine": undefined,
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          outlineButton
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Button <DoubleButton /> renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 15,
          "justifyContent": "center",
          "width": "100%",
        },
        {},
      ]
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 40,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
        >
          <View
            dataSet={{}}
            pointerEvents="auto"
            style={
              {
                "borderColor": {
                  "100": "#fee2e2",
                  "200": "#fecaca",
                  "300": "#fca5a5",
                  "400": "#f87171",
                  "50": "#fef2f2",
                  "500": "#ef4444",
                  "600": "#dc2626",
                  "700": "#b91c1c",
                  "800": "#991b1b",
                  "900": "#7f1d1d",
                },
                "borderRadius": 15,
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="blue"
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="auto"
              style={
                [
                  {
                    "alignItems": "center",
                    "borderBottomLeftRadius": 15,
                    "borderColor": "#404040",
                    "borderRadius": 0,
                    "borderTopLeftRadius": 15,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                    "width": "50%",
                  },
                  {
                    "borderColor": "#123456",
                    "borderWidth": 2,
                  },
                ]
              }
              testID="test-left"
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={{}}
                  style={{}}
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#3b82f6",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    left
                  </Text>
                </View>
              </View>
            </View>
            <View
              dataSet={{}}
              style={
                {
                  "width": 0,
                }
              }
            />
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="blue"
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="auto"
              style={
                [
                  {
                    "alignItems": "center",
                    "borderBottomRightRadius": 15,
                    "borderColor": "#404040",
                    "borderLeftWidth": 0,
                    "borderRadius": 0,
                    "borderTopRightRadius": 15,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                    "width": "50%",
                  },
                  {
                    "borderColor": "#123456",
                    "borderWidth": 2,
                  },
                ]
              }
              testID="test-right"
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={{}}
                  style={{}}
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#3b82f6",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    right
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
